dev/NavigationView/readme.md

```diff
@@ -0,0 +1,51 @@
+# NavigationView Control
+
+NavigationView is a powerful control that makes it easy to integrate a rich navigation experience into apps.
+This document outlines overall structure and behaviors.
+Information related to the NavigationViewItem, or the rendering of the control see the documents in the /NavigationView/docs folder.
+
+## Architecture
+
+NavigationView is a complex control, which has a lot of internal states to store.
+
+Rendering of the menu items is being done through three ItemsRepeaters, one for the NavigationView in left mode, one for the NavigationView in top mode and one for the overflow menu.
+For more on this, see [the rendering document](./docs/rendering.md).
+
+The selection is being tracked through a SelectionModel, which uses IndexPath tracking for hierarchical NavigationViewItems.
+Every NavigationViewItem can be selected by either setting the `IsSelected` property of a NavigationViewItem, or setting the selected item.
+To allow this, every NavigationViewItem is aware of its parent NavigationView.
+
+Keyboard navigation is done by the NavigationView, however handling of ItemInvokation is being handled by both the NavigationView and the NavigationViewItem.
```

Comment:
As far as I know, we are listening to all the item invocation events only in NavigationView. Can you point me to the code you are referring to here?

Reply:
NavigationViewItem listens to a lot of pointer events, though they are all about visuals. I'll remove NVI from that statement as it might be confusing or misleading.